src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * Automatically enters the scene that triggers when player passes out from their collar or gets brought back by enforcers.
+	 * 
+	 * It should handle the movement to a new location as well.
+	 */
+	public void triggerSlaveRetrivalDialogue()
+	{
+		//Main.game.getPlayer().setLocation(this.worldLocation, this.location, false);
+		//DialogueNodeOld dn = null; //todo
+		
+		// Warning: We are doing dangerous shit there. Since slavery system is supposed to be an overbearing one, it's safe to do it here, but to other contributors, here's a note - don't do it without a good reason!
+		// To explain - this function below sets the dialogue node.
+		// It can force it in the middle of anything, potentially sequence breaking a quest dialogue.
+		// Using it outside of these tight constraints should be avoided.
+		// (Note for Inno: I'm just leaving this as a warning for future contributors :3)
+		//Main.game.setContent(new Response("", "", dn));		
+	}
+	
+	/**
+	 * Returns true if character is wearing a clothing piece enchanted with enslaving or contraband enslaving enchantmnet.
+	 * @return
+	 */
+	public boolean isWearingEnslavingClothing()
+	{
+		return this.getClothingCurrentlyEquipped().stream().anyMatch(e -> e.isContrabandEnslavementClothing() || e.isEnslavementClothing());
+	}
+	
+	/**
+	 * Override to change the events the NPC will do to the player.
+	 */
+	public Set<PlayerSlaveryEvent> getSlaveryEvents()
+	{
+		return null;
+	}
+	
+	/**
+	 * Returns a parser-friendly remark by NPC on player's performance according to their rules.
+	 * 
+	 * Will erase the last outstanding performance remark. 
+	 * 
+	 * Returns null if there's nothing to say.
+	 * @param allowNegative Allows negative performance remarks
+	 * @param allowPositive Allows positive performance remarks
+	 * @param maxRemarks Maximum remarks before stopping the loop and returning the value
+	 * @return
+	 */
+	public String getPerformanceRemark(boolean allowNegative, boolean allowPositive, int maxRemarks)
+	{
+		String returnable = "";
+		String toAdd = null;
+		boolean willConnectLastWithAnd = false;
+		int remainingRemarks = maxRemarks;
+		
+		if(this.outstandingPerformanceRemark != null)
+		{
+			if(this.outstandingPerformanceIsNegative == allowNegative)
+			{
+				returnable = this.outstandingPerformanceRemark;
+				this.outstandingPerformanceRemark = null;
+				return returnable;
+			}
+		}
+		
+		for(PlayerSlaveryRule rule : this.slaveOwnerRules)
+		{
+			float performance = rule.getPerformanceQuality();
+			if(((performance > 0 && allowPositive) || (performance < 0 && allowNegative)) && remainingRemarks > 0)
+			{
+				if(toAdd == null)
+				{
+					toAdd = rule.getPerformanceComment();
+				}
+				else
+				{
+					willConnectLastWithAnd = true;
+					returnable += ", " + toAdd;
```

Comment:
Definitely not okay! With inputs of "did this good" "did that good" "also good stuff" it will make something like

Master said, "You , did this good, did that good and also good stuff"

Fixed that, and as a consequence another thing.